contracts/test/ClaimTest.t.sol

```diff
@@ -0,0 +1,609 @@
+
+        uint256[7] memory details = pool.claim(address(loanV2), address(debtLockerFactory));
+
+        assertEq(usdc.balanceOf(address(loanV2)), 0);
+
+        assertEq(details[0], totalPaid);
+        assertEq(details[1], interestPortion);
+        assertEq(details[2], principalPortion);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 0);
+        assertEq(details[6], 0);
+
+        assertPoolState(principalPortion, interestPortion);
+    }
+
+    function test_claim_onTimePartiallyAmortized() external {
+        createLoan(uint256(1_000_000 * USD), uint256(500_000 * USD));
+        fundLoan();
+        drawdownLoan();
+
+        // Make a single on time payment
+        (uint256 principalPortion, uint256 interestPortion) = makeLoanPayments(1,false);
+        uint256 totalPaid = principalPortion + interestPortion;
+
+        uint256[7] memory details = pool.claim(address(loanV2), address(debtLockerFactory));
+
+        assertEq(usdc.balanceOf(address(loanV2)), 0);
+
+        assertEq(details[0], totalPaid);
+        assertEq(details[1], interestPortion);
+        assertEq(details[2], principalPortion);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 0);
+        assertEq(details[6], 0);
+
+        assertPoolState(principalPortion, interestPortion);
+
+        // Make another payment
+        ( principalPortion, interestPortion ) = makeLoanPayments(1,false);
+        totalPaid = principalPortion + interestPortion;
+
+        details = pool.claim(address(loanV2), address(debtLockerFactory));
+
+        assertEq(usdc.balanceOf(address(loanV2)), 0);
+
+        assertEq(details[0], totalPaid);
+        assertEq(details[1], interestPortion);
+        assertEq(details[2], principalPortion);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 0);
+        assertEq(details[6], 0);
+
+        assertPoolState(principalPortion, interestPortion);
+
+        // Make last payment
+        ( principalPortion, interestPortion ) = makeLoanPayments(1,false);
+        totalPaid = principalPortion + interestPortion;
+
+        details = pool.claim(address(loanV2), address(debtLockerFactory));
+
+        assertEq(usdc.balanceOf(address(loanV2)), 0);
+
+        assertEq(details[0], totalPaid);
+        assertEq(details[1], interestPortion);
+        assertEq(details[2], principalPortion);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 0);
+        assertEq(details[6], 0);
+
+        assertPoolState(principalPortion, interestPortion);
+    }
+
+    function test_claim_onTimePartiallyAmortizedSingleClaim() external {
+        createLoan(uint256(1_000_000 * USD), uint256(500_000 * USD));
+        fundLoan();
+        drawdownLoan();
+
+        // Makeall three payments
+        (uint256 principalPortion, uint256 interestPortion) = makeLoanPayments(3,false);
```

Comment:
`( uint256 principalPortion, uint256 interestPortion ) = makeLoanPayments(3, false);` Same for all of these